Fix scrolling sticking to top

WeetHet opened the pull request:
I've reverted a bunch of stuff from the faulty PR, it seems to fix the problem. I don't know what exactly fixed it and honestly, I don't even care anymore

Fixes #13720.

Release Notes:

- Fixed scrolling sticking to top and bottom

WeetHet committed:
Fix #13720

maxdeviant commented:
> I've reverted a bunch of stuff from the faulty PR

What is the faulty PR? Should we revert the whole PR instead of just parts of it?

> I don't know what exactly fixed it and honestly, I don't even care anymore

This is not satisfactory. If we don’t understand the problem we can’t be making changes that “fix” it.

WeetHet commented:
> > I've reverted a bunch of stuff from the faulty PR
> 
> What is the faulty PR? Should we revert the whole PR instead of just parts of it?
> 
> > I don't know what exactly fixed it and honestly, I don't even care anymore
> 
> This is not satisfactory. If we don’t understand the problem we can’t be making changes that “fix” it.

I know that #13368 broke scrolling and I know it's something in the two files I've changed. I have just `git restore`-d them and reapplied the changes that didn't break scrolling

WeetHet commented:
> > I don't know what exactly fixed it and honestly, I don't even care anymore
> 
> This is not satisfactory. If we don’t understand the problem we can’t be making changes that “fix” it.
Sure, I don't have time to investigate this even further right now, so I'll be closing this PR
